src/Atmos/Model/AtmosModel.jl

```diff
@@ -223,8 +228,12 @@ struct NoFluxBC <: BoundaryCondition
 end
 function boundarycondition!(m::AtmosModel{O,T,M,R,S,BC,IS}, stateP::Vars, diffP::Vars, auxP::Vars,
     nM, stateM::Vars, diffM::Vars, auxM::Vars, bctype, t) where {O,T,M,R,S,BC <: NoFluxBC,IS}
-
-  stateP.ρu -= 2 * dot(stateM.ρu, nM) * nM
+    DF = eltype(stateM)
+    UM, VM, WM = stateM.ρu
+    stateP.ρ = stateM.ρ
+    stateP.ρu -= 2 * dot(stateM.ρu, nM) * collect(nM)
```

Comment:
I think that this layer should be invisible for the user who is not familiar with the mathematics of b.c.
He should only see something like "no-flux", "heat flux", "periodic", etc. Based on this, the b.c. are defined in the background.
What makes it even more difficult for the user not familiar with DG  is the fact that we assign a `Plus` state as a function of the `Minus` state. A user should never have to wonder what this means.

Reply:
That's the plan, we're slowly in the process of hiding more of this.